pupil-spa/src/app/services/storage/storage.service.ts

```diff
@@ -1,17 +1,187 @@
 import { Injectable } from '@angular/core';
-import { StorageKeyTypesAll, StorageKeyPrefix } from './storageKey';
+import {
+  StorageKeyTypesAll,
+  AnswersStorageKey,
+  AccessArrangementsStorageKey,
+  PupilStorageKey,
+  CompletedSubmissionStorageKey,
+  TimeoutStorageKey,
+  ConfigStorageKey,
+  CheckStateStorageKey,
+  PendingSubmissionStorageKey,
+  QuestionsStorageKey,
+  FeedbackStorageKey,
+  SchoolStorageKey,
+  DeviceStorageKey,
+  AuditStorageKey,
+  AccessTokenStorageKey,
+  InputsStorageKey,
+  CheckStartTimeStorageKey,
+  TokensStorageKey
+} from './storageKey';
+import { Answer } from '../answer/answer.model';
+import { AccessArrangements } from '../../access-arrangements';
+import { AuditEntry } from '../audit/auditEntry';
+import { Question } from '../question/question.model';
+
```

Comment:
💯 
we will need to make sure that subsequent takes of the check are not affected by using the same key instance, for these particular items, as they would not be newed up again on a retake.  
although i do see that you are explicitly newing up other specific keys each time they are added, which is good.
maybe we need to put these calls in a method, and call it explicitly after they login, as part of setup?
happy to discuss, as i may be overthinking this